Stock.Api/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Stock.Api.DTOs;
+using Stock.Api.Extensions;
+using Stock.AppService.Services;
+using Stock.Model.Entities;
+
+namespace Stock.Api.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/provider")]
+    [ApiController]
+    public class ProviderController : ControllerBase
+    {
+        private ProviderService service;
+        private readonly IMapper mapper;
+
+        public ProviderController(ProviderService service, IMapper mapper)
+        {
+            this.service = service;
+            this.mapper = mapper;
+        }
+
+        [HttpPost]
+        public ActionResult Post([FromBody] ProviderDTO value)
+        {
+            TryValidateModel(value);
+
+            try
+            {
+                var provider = this.mapper.Map<Provider>(value);
+                this.service.Create(provider);
+                value.Id = provider.Id;
+                return Ok(new { Success = true, Message = "", data = value });
+            }
+            catch
```

Comment:
No está bueno usar excepciones "anónimas" ni esos catchs genéricos por defecto.

Si se usan excepciones, deberían ser nombradas y además deberías tratarlas: 

`catch(AlgunaException e){ /*tarea pertinente*/ }`

Recomiendo ver [este link](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions)